### PR TITLE
Update README and Developer Guide

### LINKED ISSUE #438: Add documentation for creating new Workflow Steps
### Is your feature request related to a problem?

Those who haven't written a workflow step don't know the different places things need to be added.

### What solution would you like?

We should add a section to the developer notes on how to do it.

### What alternatives have you considered?

Waiting for people to ask for help, or searching various "add X step" PRs

### PR DESCRIPTION
**Pull request by dbwiddis**
### Description

 - Updates the README with more context and links to our incremental development issue.
 - Adds a section to the Developer Guide identifying how to create new Workflow Steps.

### Issues Resolved

Resolves #438 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by jackiehanyang — APPROVED** [review]

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -93,8 +97,16 @@ snapshots/
...
+
 1. Change the url from ``"https://aws.oss.sonatype.org/content/repositories/snapshots"`` to your local path and comment out the credentials under publishing/repositories in build.gradle.
 2. Run ```./gradlew publishPluginZipPublicationToSnapshotsRepository```.
+
+### Adding Workflow Steps
+
+To add functionality to workflows, add new Workflow Steps to the [`org.opensearch.flowframework.workflow`](https://github.com/opensearch-project/flow-framework/tree/main/src/main/java/org/opensearch/flowframework/workflow) package.
+1. Implement the [Workflow](https://github.com/opensearch-project/flow-framework/blob/main/src/main/java/org/opensearch/flowframework/workflow/WorkflowStep.java) interface. See existing steps for examples for input, output, and API execution.
+2. Choose a unique name for the step which is not used by other steps.
+3. Add a constructor and call it from the [WorkflowStepFactory](https://github.com/opensearch-project/flow-framework/blob/main/src/main/java/org/opensearch/flowframework/workflow/WorkflowStepFactory.java).
```
Comment (amitgalitz) [inline DEVELOPER_GUIDE.md:110]: Should we add a step regarding the resources section if their steps creates a resource?

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -93,8 +97,16 @@ snapshots/
...
-
 ```
+
 1. Change the url from ``"https://aws.oss.sonatype.org/content/repositories/snapshots"`` to your local path and comment out the credentials under publishing/repositories in build.gradle.
 2. Run ```./gradlew publishPluginZipPublicationToSnapshotsRepository```.
+
+### Adding Workflow Steps
+
+To add functionality to workflows, add new Workflow Steps to the [`org.opensearch.flowframework.workflow`](https://github.com/opensearch-project/flow-framework/tree/main/src/main/java/org/opensearch/flowframework/workflow) package.
+1. Implement the [Workflow](https://github.com/opensearch-project/flow-framework/blob/main/src/main/java/org/opensearch/flowframework/workflow/WorkflowStep.java) interface. See existing steps for examples for input, output, and API execution.
```
Comment (amitgalitz) [inline DEVELOPER_GUIDE.md:108]: Should we add a link to an existing example like https://github.com/opensearch-project/flow-framework/blob/main/src/main/java/org/opensearch/flowframework/workflow/CreateConnectorStep.java so they know what to reference?

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (dbwiddis) [reply DEVELOPER_GUIDE.md:110]: Oops! Yes, I had that thought and then that thought went away...

File: DEVELOPER_GUIDE.md (hunk 2 above)
Comment (dbwiddis) [reply DEVELOPER_GUIDE.md:108]: I linked to the package folder with lots of examples so I think that's probably good enough.

**Review by amitgalitz — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+We want to introduce our customers to a new no-code/low-code builder experience ([Backend RFC](https://github.com/opensearch-project/OpenSearch/issues/9213) and [Frontend RFC](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4755)) that empowers users to compose AI-augmented query and ingestion flows, integrate ML models supported by ML-Commons, and streamline the OpenSearch app development experience through a drag-and-drop designer.  The front end will help users create use case templates, which provide a compact description of configuration steps for automated workflows such as Retrieval Augment Generation (RAG), AI connectors and other components that prime OpenSearch as a backend to leverage generative models.  Once primed, builders can query OpenSearch directly without building middleware logic to stitch together data flows and ML models.
```
Comment (amitgalitz) [inline README.md:7]: Should we add a statement that this is more then just ML and can power many different workflows in opensearch, relating to alerts, other plugins, observability and etc to make it clear we don't want to only onboard AI/ML use cases.

**Pull request merged**